Mathlib/Order/Lattice.lean

```diff
@@ -902,9 +902,27 @@ theorem map_inf_le [SemilatticeInf α] [SemilatticeInf β] {f : α → β} (h :
     f (x ⊓ y) ≤ f x ⊓ f y :=
   le_inf (h inf_le_left) (h inf_le_right)
 
+theorem of_map_inf_le_left [SemilatticeInf α] [Preorder β] {f : α → β}
+    (h : ∀ x y, f (x ⊓ y) ≤ f x) : Monotone f := by
+  intro x y hxy
+  rw [← inf_eq_right.2 hxy]
+  apply h
+
+theorem of_map_inf_le [SemilatticeInf α] [SemilatticeInf β] {f : α → β}
+    (h : ∀ x y, f (x ⊓ y) ≤ f x ⊓ f y) : Monotone f :=
+  of_map_inf_le_left fun x y ↦ (h x y).trans inf_le_left
+
 theorem of_map_inf [SemilatticeInf α] [SemilatticeInf β] {f : α → β}
     (h : ∀ x y, f (x ⊓ y) = f x ⊓ f y) : Monotone f :=
-  fun x y hxy => inf_eq_left.1 <| by rw [← h, inf_eq_left.2 hxy]
+  of_map_inf_le fun x y ↦ (h x y).le
```

Comment:
Is this one still worth keeping, then? If not, please deprecate

Reply:
This is a natural condition. It is used twice in the library. In both cases, it's possible to use `of_map_inf_le` but it turns `of_map_inf h` into `of_map_inf_le fun _ _ ↦ (h _ _).le`.